packages/react-components/react-toast/src/state/vanilla/toaster.ts

```diff
@@ -0,0 +1,80 @@
+import { Toast, ToastEventMap, ToastId, ToastOptions } from '../types';
+import { EVENTS } from '../constants';
+
+// TODO convert to closure
+export class Toaster {
+  public visibleToasts: Set<ToastId>;
+  public toasts: Map<ToastId, Toast>;
+  public onUpdate: () => void;
+  private targetDocument: Document;
+
+  private listeners = new Map<keyof ToastEventMap, Function>();
```

Comment:
I would be more strict and introduce `ToastEventListener` like `type ToastEventListener = <K extends keyof ToastEventMap>(e: ToastEventMap[K]) => void`. That will cleanup some of type casts